tests/common/dualtor/dual_tor_mock.py

```diff
@@ -0,0 +1,338 @@
+        swss_filename: The filename to use for copying the config file around (default='swss_config_file')
+    '''
+
+    dut_filename = '/tmp{}'.format(swss_filename)
+
+    dut.shell('echo "{}" > {}'.format(swss_config_str, dut_filename))
+    dut.shell('docker cp {} swss:{}'.format(dut_filename, swss_filename))
+    dut.shell('docker exec swss sh -c "swssconfig {}"'.format(swss_filename))
+
+
+def _apply_dual_tor_state_to_orchagent(dut, state):
+    '''
+    Helper function to configure active/standby state in orchagent
+
+    Args:
+        dut: DUT object
+        state: either 'active' or 'standby'
+    '''
+
+    logger.info("Applying {} state to orchagent".format(state))
+
+    vlan_intfs = sorted(dut.get_vlan_intfs(), key=lambda intf: int(intf.replace('Ethernet', '')))
+
+    intf_configs = []
+
+    for intf in vlan_intfs:
+        '''
+        For each VLAN interface, create one configuration to be applied to orchagent
+        Each interface configuration has the following structure:
+
+        {
+            "MUX_CABLE_TABLE:<intf name>": {
+                "state": <active/standby>
+            }
+            "OP": "SET"
+        }
+        '''
+        intf_config_dict = {}
+        state_dict = {}
+
+        state_key = '"MUX_CABLE_TABLE:{}"'.format(intf)
+        state_dict = {'"state"': '"{}"'.format(state)}
+        intf_config_dict[state_key] = state_dict
+        intf_config_dict['"OP"'] = '"SET"'
+
+        intf_configs.append(intf_config_dict)
+
+    swss_config_str = json.dumps(intf_configs, indent=4)
+    logger.debug('SWSS config string is {}'.format(swss_config_str))
+    swss_filename = '/mux{}.json'.format(state)
+    _copy_config_to_swss(dut, swss_config_str, swss_filename)
+
+    yield
+    logger.info("Removing {} state from orchagent".format(state))
+
+    for i in range(len(intf_configs)):
+        intf_configs[i]['"OP"'] = '"DEL"'
+
+    swss_config_str = json.dumps(intf_configs, indent=4)
+    swss_filename = '/mux{}.json'.format(state)
+    _copy_config_to_swss(dut, swss_config_str, swss_filename)
+
+
+@pytest.fixture(scope='module')
+def apply_active_state_to_orchagent(duthosts, rand_one_dut_hostname):
+    dut = duthosts[rand_one_dut_hostname] 
+
+    for func in _apply_dual_tor_state_to_orchagent(dut, 'active'):
+        yield func
+
+
+@pytest.fixture(scope='module')
+def apply_standby_state_to_orchagent(duthosts, rand_one_dut_hostname):
+    dut = duthosts[rand_one_dut_hostname]
+
+    for func in _apply_dual_tor_state_to_orchagent(dut, 'standby'):
+        yield func
+
+
+@pytest.fixture(scope='module')
+def mock_peer_switch_loopback_ip(duthosts, rand_one_dut_hostname):
+    '''
+    Returns the mocked peer switch loopback IP
+
+    The peer switch loopback is always the next IP address after the DUT loopback
+
+    Returns:
+        IPv4Interface object
+    '''
+
+    dut = duthosts[rand_one_dut_hostname]
+    lo_facts = dut.get_running_config_facts()['LOOPBACK_INTERFACE']
```

Comment:
Just to confirm, is this picking up `Loopback0` ip?

Reply:
Just whichever is listed first. If the config came from a minigraph generated by sonic-mgmt, it will always be Loopback0. Also, the single ToR config only has a single Loopback interface regardless.